src/annotator/cross-frame.js

```diff
@@ -55,70 +56,87 @@ export class CrossFrame {
     };
 
-    frameObserver.observe(injectIntoFrame, iframeUnloaded);
+    this._frameObserver.observe(injectIntoFrame, iframeUnloaded);
+  }
 
-    /**
-     * Attempt to connect to the sidebar frame.
-     *
-     * Returns a promise that resolves once the connection has been established.
-     *
-     * @param {Window} frame - The window containing the sidebar application
-     * @param {string} origin - Origin of the sidebar application (eg. 'https://hypothes.is/')
-     */
-    this.connectToSidebar = (frame, origin) => {
-      const channel = new MessageChannel();
-      frame.postMessage(
-        {
-          type: 'hypothesisGuestReady',
-        },
-        origin,
-        [channel.port2]
-      );
-      bridge.createChannel(channel.port1);
-    };
+  /**
+   * Attempt to connect to the sidebar frame.
+   *
+   * Returns a promise that resolves once the connection has been established.
+   *
+   * @param {Window} frame - The window containing the sidebar application
+   * @param {string} origin - Origin of the sidebar application (eg. 'https://hypothes.is/')
+   */
+  connectToSidebar(frame, origin) {
+    const channel = new MessageChannel();
+    frame.postMessage(
+      {
+        type: 'hypothesisGuestReady',
+      },
+      origin,
+      [channel.port2]
+    );
+    this._bridge.createChannel(channel.port1);
+  }
 
-    /**
-     * Remove the connection between the sidebar and annotator.
-     */
-    this.destroy = () => {
-      bridge.destroy();
-      annotationSync.destroy();
-      frameObserver.disconnect();
-    };
+  /**
+   * Remove the connection between the sidebar and annotator.
+   */
+  destroy() {
+    this._bridge.destroy();
+    this._annotationSync.destroy();
+    this._frameObserver.disconnect();
+  }
 
-    /**
-     * Notify the sidebar about new annotations created in the page.
-     *
-     * @param {AnnotationData[]} annotations
-     */
-    this.sync = annotations => annotationSync.sync(annotations);
+  /**
+   * Notify the sidebar about new annotations created in the page.
+   *
+   * @param {AnnotationData[]} annotations
+   */
+  sync(annotations) {
+    this._annotationSync.sync(annotations);
+  }
 
-    /**
-     * Subscribe to an event from the sidebar.
-     *
-     * @param {string} event
-     * @param {(...args: any[]) => void} callback
-     */
-    this.on = (event, callback) => bridge.on(event, callback);
+  /**
+   * Subscribe to an event from the sidebar.
+   *
+   * @param {string} method
+   * @param {(...args: any[]) => void} listener -- Final argument is an optional
+   *   callback of the type: `(error: string|Error|null, ...result: any[]) => void`.
+   *   This callback must be invoked in order to respond (via `postMessage`)
+   *   to the other frame/s with a result or an error.
+   * @throws {Error} If trying to register a callback after a channel has already been created
+   * @throws {Error} If trying to register a callback with the same name multiple times
```

Comment:
Rather than repeating the documentation from the `Bridge#on` method, it would probably be safer to just write something like: 

```js
/**
 * Subscribe to an event from the sidebar.
 *
 * @see {Bridge.on} for details.
 */
```

This way the documentation won't get out of date if `Bridge.on` changes in future.

Reply:
Implemented in commit number 4.

The only problem with this approach is that we lose the types for the parameters.